### ISSUE TITLE
generating job from ros_buildfarm checkout embed wrong path to scripts in jobs

### ISSUE THREAD

**Issue by asmodehn** [ros-infrastructure/ros_buildfarm#101]
I followed the instructions to setup ros_buildfarm, and as the released one (0.2.0) failed to find the scripts in bin ( I already saw https://github.com/ros-infrastructure/ros_buildfarm/issues/80 about this problem ),  I opted for the cloned repository option.

Now, when I generate all the jobs ( keeping https://github.com/ros-infrastructure/ros_buildfarm/issues/85 in mind ) everything looks fine : 
```
(ros_buildfarm)alexv@alexv-PC:/tmp/deploy_ros_buildfarm/ros_buildfarm/scripts$ ./generate_all_jobs.py file:///home/alexv/Projects/ros_buildfarm_config/index.yaml
Connecting to Jenkins 'http://asmodehn:8080'
Connected to Jenkins version '1.630'

Invoking 'misc/generate_check_slaves_job.py file:///home/alexv/Projects/ros_buildfarm_config/index.yaml'
[...]
Invoking 'status/generate_repos_status_page_job.py file:///home/alexv/Projects/ros_buildfarm_config/index.yaml jade'

Connecting to Jenkins 'http://asmodehn:8080'
Connected to Jenkins version '1.630'
Ensure that view 'Manage' exists
Skipped 'Manage' because the config is the same
Creating job 'Jrel_repos-status-page'
The Jenkins master does not require a crumb
```
but when I inspect the job configuration I find this : 

```
# monitor all subprocesses and enforce termination
python3 -u $WORKSPACE/ros_buildfarm/scripts/subprocess_reaper.py $$ --cid-file $WORKSPACE/docker_generate_rosdistro_cache/docker.cid > $WORKSPACE/docker_generate_rosdistro_cache/subprocess_reaper.log 2>&1 &
# sleep to give python time to startup
sleep 1
```

which sure enough fails to find the script : 
```
+ export PYTHONPATH=/var/lib/jenkins/workspace/indigo_rosdistro-cache/ros_buildfarm:
+ python3 -u /var/lib/jenkins/workspace/indigo_rosdistro-cache/ros_buildfarm/scripts/misc/run_rosdistro_cache_job.py --rosdistro-index-url file:///home/alexv/Projects/rosdistro-yujin/index.yaml indigo --distribution-repository-urls http://asmodehn/ubuntu/building --distribution-repository-key-files /var/lib/jenkins/workspace/indigo_rosdistro-cache/keys/0.key --dockerfile-dir /var/lib/jenkins/workspace/indigo_rosdistro-cache/docker_generate_rosdistro_cache
python3: can't open file '/var/lib/jenkins/workspace/indigo_rosdistro-cache/ros_buildfarm/scripts/misc/run_rosdistro_cache_job.py': [Errno 2] No such file or directory
```

as it is in `/tmp/deploy_ros_buildfarm/ros_buildfarm/scripts/misc/run_rosdistro_cache_job.py`

Let me know if I did anything obviously wrong, or where the issue might be ( scripts, templates, need to copy ros_buildfarm/scripts into workspace ? ), I m willing to make a pull request for this.
Cheers.

**Comment by tfoote** [comment]
The job should be checking out it's own copy of the repo. The jobs should have no knowledge of the path to where you ran the configuration script as it may not be local to the jenkins executor.

Can you look in the workspace and find out what version of ros_buildfarm is checked out?

**Comment by asmodehn** [comment]
Oh, thats interesting...

The ros_buildfarm version I cloned manually, and I used to generate the jobs : 
```
(ros_buildfarm)alexv@alexv-PC:/tmp/deploy_ros_buildfarm/ros_buildfarm/scripts$ git log -n 1
commit 8ebdfad3b3c45042bb749a7f568018d91d440116
Author: Tully Foote <tfoote@osrfoundation.org>
Date:   Tue Sep 1 16:57:13 2015 -0700

    add section on calling import upstream
```

I dont think I have any ros_buildfarm checked out in the workspace. 

```
(ros_buildfarm)alexv@alexv-PC:/var/lib/jenkins/workspace$ find . -name ros_buildfarm
(ros_buildfarm)alexv@alexv-PC:/var/lib/jenkins/workspace$ ls -al
total 40
drwxr-xr-x 10 jenkins jenkins 4096  9월 24 23:01 .
drwxr-xr-x 11 jenkins jenkins 4096  9월 25 08:47 ..
drwxr-xr-x  2 jenkins jenkins 4096  9월 24 17:11 check_slaves
drwxr-xr-x  2 jenkins jenkins 4096  9월 24 18:00 dashboard
drwxr-xr-x  4 jenkins jenkins 4096  9월 25 10:30 indigo_rosdistro-cache
drwxr-xr-x  2 jenkins jenkins 4096  9월 24 23:01 Irel_arm_trigger-broken-with-non-broken-upstream
drwxr-xr-x  2 jenkins jenkins 4096  9월 24 23:01 Irel_trigger-broken-with-non-broken-upstream
drwxr-xr-x  4 jenkins jenkins 4096  9월 25 10:30 jade_rosdistro-cache
drwxr-xr-x  2 jenkins jenkins 4096  9월 24 23:01 Jrel_arm_trigger-broken-with-non-broken-upstream
drwxr-xr-x  2 jenkins jenkins 4096  9월 24 23:01 Jrel_trigger-broken-with-non-broken-upstream
(ros_buildfarm)alexv@alexv-PC:/var/lib/jenkins/workspace$ 
```

Also the job configuration does not seem to have any repository configured, at least form the jenkins configuration webpage : Source Code Management -> None.
My jenkins version is 1.630.

But Jenkins is telling me I have old data : 
```
hudson.model.FreeStyleProject	indigo_rosdistro-cache	CannotResolveClassException: jenkins.advancedqueue.AdvancedQueueSorterJobProperty, CannotResolveClassException: org.jenkinsci.plugins.requeuejob.RequeueJobProperty, CannotResolveClassException: hudson.plugins.git.GitSCM, CannotResolveClassException: jenkins.plugins.publish_over_ssh.BapSshPublisherPlugin, CannotResolveClassException: hudson.plugins.timestamper.TimestamperBuildWrapper
```

I guess there is a list of plugin that I must install with jenkins for ros buildfarm to work ?

**Comment by asmodehn** [comment]
So I installed : 

* GIT plugin	 ( hudson.plugins.git.GitSCM )
* Requeue Job Plugin	 ( org.jenkinsci.plugins.requeuejob.RequeueJobProperty )
* Timestamper	 ( hudson.plugins.timestamper.TimestamperBuildWrapper )
* Publish Over SSH	 ( jenkins.plugins.publish_over_ssh.BapSshPublisherPlugin )
* Priority Sorter Plugin	 ( jenkins.advancedqueue.AdvancedQueueSorterJobProperty )

And I could see the rosdistro job cloning ros_buildfarm.

I also needed to add : 

* Groovy (hudson.plugins.groovy.SystemGroovy ) 
* build timeout (hudson.plugins.build_timeout.BuildTimeoutWrapper
* SSH Agent Plugin ( com.cloudbees.jenkins.plugins.sshagent.SSHAgentBuildWrapper )

These should probably be in a doc somewhere I guess

**Comment by tfoote** [comment]
It sounds like your master failed to successfully finish it's puppet configuration. All those plugins are necessary, and would explain why you don't get the script in your workspace. You should rerun the puppet configuration for the master and check for any error outputs. 

There should be a lot of plugins installed automatically for you. [see here](https://github.com/ros-infrastructure/buildfarm_deployment/blob/master/master/manifests/site.pp#L162-L293)

**Comment by asmodehn** [comment]
The main point I think, is that I didn't expect to have to use that whole puppet deployment mechanism just to setup a build server.

After all, my first step in this whole automated build setup was to get my custom rosdistro repo and build configuration right, before starting to play around with setting up multiple build machines.

So I just installed jenkins, and checkedout ros_buildfarm, and generated the jobs... It s only after that I realized I needed to have a more complex setup before being able to run the buildfarm.

**Comment by tfoote** [comment]
The main documentation entry point is here: http://wiki.ros.org/buildfarm
You do need the full setup of all the plugins to be able to run automation of which is here: https://github.com/ros-infrastructure/buildfarm_deployment

We can't get all that configuration setup into the job generation so I'm going to close this as won't fix.
